### PR TITLE
[MLU] add any, add square&square kernel test, fix reduce_op

### PR DESCRIPTION
**Pull request by PeiyuLau**

### CODE REVIEW

**Comment by PeiyuLau** [comment]
square:
image

**Comment by PeiyuLau** [comment]
reduce_any:
image

**Comment by ShawnNew** [comment]
do not merge until [pr-626](https://github.com/PaddlePaddle/PaddleCustomDevice/pull/626)

File: backends/mlu/kernels/reduce_any_kernel.cc [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+  // broadcast
+  std::vector<int64_t> dim_vec = dims;
+  auto x_dims_vec = phi::vectorize(x.dims());
+  if (reduce_all) {
+    dim_vec.clear();
+    for (size_t d = 0; d < x_dims_vec.size(); ++d) {
+      dim_vec.push_back(static_cast<int64_t>(d));
+    }
+  }
+  MLUReduceOp<T>(dev_ctx, x, dim_vec, keep_dim, reduce_all, "reduce_any", out);
```
Comment (ShawnNew) [inline backends/mlu/kernels/reduce_any_kernel.cc]: the upper logic may included in this function already, minor changes are suggested.

**Review by ShawnNew — CHANGES_REQUESTED** [review]

**Review by ShawnNew — APPROVED** [review]
LGTM

**Pull request merged**